lib/ocpp/v201/connectivity_manager.cpp

```diff
@@ -131,15 +196,50 @@ void ConnectivityManager::init_websocket() {
     cache_network_connection_profiles();
 
     const auto configuration_slot = this->network_connection_priorities.at(this->network_configuration_priority);
-    const auto connection_options = this->get_ws_connection_options(std::stoi(configuration_slot));
-
-    const auto network_connection_profile = this->get_network_connection_profile(std::stoi(configuration_slot));
+    const int config_slot_int = std::stoi(configuration_slot);
+
+    const auto network_connection_profile = this->get_network_connection_profile(config_slot_int);
+    // Not const as the iface member can be set by the configure network connection profile callback
+    auto connection_options = this->get_ws_connection_options(config_slot_int);
+    bool can_use_connection_profile = true;
+
+    if (!network_connection_profile.has_value()) {
+        EVLOG_warning << "No network connection profile configured for " << config_slot_int;
+        can_use_connection_profile = false;
+    } else if (this->configure_network_connection_profile_callback.has_value()) {
+        EVLOG_debug << "Request to configure network connection profile " << config_slot_int;
+
+        std::future<ConfigNetworkResult> config_status = this->configure_network_connection_profile_callback.value()(
```

Comment:
Might be good to split this off to a function to handle the callback so this reads a bit cleaner